legacy/src/main/java/org/opensearch/sql/legacy/executor/multi/MinusExecutor.java

```diff
@@ -63,45 +72,63 @@ public MinusExecutor(Client client, MultiQueryRequestBuilder builder) {
 
   @Override
   public void run() throws SqlParseException {
-    if (this.useTermsOptimization && this.fieldsOrderFirstTable.length != 1) {
-      throw new SqlParseException(
-          "Terms optimization failed: terms optimization for minus execution is supported with one"
-              + " field");
-    }
-    if (this.useTermsOptimization && !this.useScrolling) {
-      throw new SqlParseException(
-          "Terms optimization failed: using scrolling is required for terms optimization");
-    }
-    if (!this.useScrolling || !this.useTermsOptimization) {
-      Set<ComperableHitResult> comperableHitResults;
-      if (!this.useScrolling) {
-        // 1. get results from first search , put in set
-        // 2. get reults from second search
-        // 2.1 for each result remove from set
-        comperableHitResults = simpleOneTimeQueryEach();
+    try {
+      if (LocalClusterState.state().getSettingValue(SQL_PAGINATION_API_SEARCH_AFTER)) {
+        pit =
+            new PointInTimeHandlerImpl(
+                client,
+                ArrayUtils.concat(
+                    builder.getOriginalSelect(true).getIndexArr(),
+                    builder.getOriginalSelect(false).getIndexArr()));
+        pit.create();
+      }
+
+      if (this.useTermsOptimization && this.fieldsOrderFirstTable.length != 1) {
```

Comment:
The logic starting from this become even more complex than before. Could you explain why and think about if any better approach?

Reply:
I see. I recall we have this similar logic in your previous PR? If all `ElasticHitsExecutor` has such logic, can we move it to parent class?

Reply:
No, not all ElasticHitsExecutor has similar logic. Join and multiquery has this pit logic but there is no scrolling in UnionExecutor so it doesn't have such logic.